packages/plugin-ext/src/plugin/types-impl.ts

```diff
@@ -2125,6 +2131,52 @@ export class DebugAdapterServer {
     }
 }
 
+export namespace DebugAdapterServer {
+    export function is(adapter: theia.DebugAdapterDescriptor | undefined): adapter is DebugAdapterServer {
+        return !!adapter && 'port' in adapter;
+    }
+}
+
+/**
+ * Represents a debug adapter running as a Named Pipe (on Windows)/UNIX Domain Socket (on non-Windows) based server.
+ */
+@es5ClassCompat
+export class DebugAdapterNamedPipeServer {
+    /**
+     * Create a description for a debug adapter running as a Named Pipe (on Windows)/UNIX Domain Socket (on non-Windows) based server.
+     */
+    constructor(readonly path: string) { }
+}
+
+export namespace DebugAdapterNamedPipeServer {
+    export function is(adapter: theia.DebugAdapterDescriptor | undefined): adapter is DebugAdapterNamedPipeServer {
+        return !!adapter && 'path' in adapter;
+    }
+}
+
+/**
+ * A debug adapter descriptor for an inline implementation.
+ */
+@es5ClassCompat
+export class DebugAdapterInlineImplementation implements DebugAdapterInlineImplementation {
```

Comment:
Shouldn't this be: 
```suggestion
export class DebugAdapterInlineImplementation {
```

Reply:
In formal logic, you'd probably call it a tautology 🤦